feat: Support removing properties from catalogs (#713)
stackabletech/trino-operator: docs/modules/trino/pages/usage-guide/catalogs/index.adoc

The `spec.connector.<connector>` determines which connector is used.
Each connector supports a different set of attributes.

=== Config overrides and config removals

You can use `.spec.configOverrides` to set arbitrary additional properties, which will be added to the catalog.

There is also `.spec.experimentalConfigRemovals` to remove any properties the operator might set, but are not used by Trino.

> This causes Trino to log an error message such as `Error: Configuration property 'hive.s3.aws-access-key' was not used` and refuse startup.

Reviewer: ```suggestion
This causes Trino to refuse to startup with an error message such as `Error: Configuration property 'hive.s3.aws-access-key' was not used`.
```
